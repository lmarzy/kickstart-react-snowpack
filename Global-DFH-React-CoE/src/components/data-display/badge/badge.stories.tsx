import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Badge } from './badge';

export default {
  title: 'Components/DataDisplay/Badge',
  component: Badge,
} as Meta;

export const Default: Story = ({ type, children }) => {
  return <Badge type={type}>{children}</Badge>;
};

Default.args = {
  children: '1',
};
