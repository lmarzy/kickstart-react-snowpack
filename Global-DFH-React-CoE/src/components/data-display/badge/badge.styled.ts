import styled from 'styled-components';

import { feedbackColors } from '../../..';
import { FeedbackType } from '../../../shared/types-styles';

interface StyledSpanProps {
  $type: FeedbackType;
}

export const StyledSpan = styled.span<StyledSpanProps>`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: var(--height-tiny);
  height: var(--height-tiny);
  font-size: var(--size-1);
  background-color: ${({ $type }) =>
    `var(--col-${feedbackColors($type).backgroundColor})`};
  border-width: 1px;
  border-style: solid;
  border-color: ${({ $type }) => `var(--col-${feedbackColors($type).borderColor})`};
  border-radius: 50%;
`;
