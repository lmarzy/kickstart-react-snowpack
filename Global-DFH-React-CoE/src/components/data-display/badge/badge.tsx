import React, { FC } from 'react';

import { FeedbackType } from '../../../shared/types-styles';

import { StyledSpan } from './badge.styled';

interface BadgeProps {
  type: FeedbackType;
}

export const Badge: FC<BadgeProps> = ({ type, children }) => (
  <StyledSpan $type={type}>{children}</StyledSpan>
);
