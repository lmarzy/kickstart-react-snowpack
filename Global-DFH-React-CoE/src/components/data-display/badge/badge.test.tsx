import React from 'react';
import { render, screen } from '@testing-library/react';

import { Badge } from './badge';

describe('Badge Component', () => {
  test('renders an info badge component', () => {
    render(<Badge type="info">2</Badge>);

    const badge = screen.getByText('2');

    expect(badge).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });

  test('renders an success badge component', () => {
    render(<Badge type="success">2</Badge>);

    const badge = screen.getByText('2');

    expect(badge).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });

  test('renders an warning badge component', () => {
    render(<Badge type="warning">2</Badge>);

    const badge = screen.getByText('2');

    expect(badge).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });

  test('renders an error badge component', () => {
    render(<Badge type="error">2</Badge>);

    const badge = screen.getByText('2');

    expect(badge).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });
});
