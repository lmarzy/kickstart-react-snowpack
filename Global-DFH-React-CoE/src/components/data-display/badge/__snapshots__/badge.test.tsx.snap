// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders an error badge component 1`] = `[Function]`;

exports[`Badge Component renders an info badge component 1`] = `[Function]`;

exports[`Badge Component renders an success badge component 1`] = `[Function]`;

exports[`Badge Component renders an warning badge component 1`] = `[Function]`;
